src/imp/signals.c

```diff
@@ -107,5 +108,16 @@ void imp_setup_signal_forwarding (struct imp_state *imp)
        imp_die (1, "failed to block signals: %s", strerror (errno));
 }
 
+void imp_raise (int sig)
+{
+    signal (sig, SIG_DFL);
+    if (raise (sig) == 0)
+        pause ();
```

Comment:
Oh, nice!  I hadn't even thought about the fact that the wait status would not have WIFSIGNALED true etc. in this case.